WalletWasabi/Crypto/CryptoGuards.cs

```diff
@@ -2,22 +2,36 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
+using NBitcoin.Secp256k1;
 using WalletWasabi.Crypto.Groups;
 
 namespace WalletWasabi.Crypto
 {
 	public static class CryptoGuard
 	{
 		[DebuggerStepThrough]
-		public static GroupElement NotInfinity(string parameterName, GroupElement groupElement)
-			=> groupElement.IsInfinity switch
+		public static GroupElement NotNullOrInfinity(string parameterName, GroupElement groupElement)
```

Comment:
You are calling this method from the other `NotNullOrInfinity`. The API would be nicer if you'd turn it around and call the other method from this one instead:

```cs
		public static GroupElement NotNullOrInfinity(string parameterName, params GroupElement[] groupElements) => NotNullOrInfinity(parameterName, groupElements as IEnumerable<GroupElement>);
```

Of course this means you have to implement the logic in the other method.

Reply:
I tried to do this because it seems good but after investing some time I was not able to do it because one method returns one `GroupElement` while the other one returns an `IEnumerable<GroupElement>`.